test/integration_test/basic_test.go

```diff
@@ -771,3 +788,86 @@ func AlertManagerRegression(tc *types.TestCase) func(*testing.T) {
+// 5. add a new label 'topology.kubernetes.io/zone' to all nodes, then validate new constraint should be added to deployments
+// 6. remove all topology labels, then validate all topology constraints should be removed
+// 7. uninstall the cluster and recover node topology labels
+func InstallWithNodeTopologyLabels(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		regionKey := "topology.kubernetes.io/region"
+		zoneKey := "topology.kubernetes.io/zone"
+		// Backup existing topology labels
+		logrus.Infof("Backing up cluster topology labels")
+		nodeList, err := coreops.Instance().GetNodes()
+		require.NoError(t, err)
+		for _, node := range nodeList.Items {
+			region := ""
+			zone := ""
+			if val, ok := node.Labels[regionKey]; ok {
+				region = val
+			}
+			if val, ok := node.Labels[zoneKey]; ok {
+				zone = val
+			}
+			logrus.Infof("node %s: %s=%s, %s=%s", node.Name, regionKey, region, zoneKey, zone)
+		}
+
+		// Add/overwrite region topology label to all nodes, divide all nodes into 2 regions
+		for i, node := range nodeList.Items {
+			val := fmt.Sprintf("region%v", i%2)
+			logrus.Infof("Label node %s with %s=%s", node.Name, regionKey, val)
+			err := coreops.Instance().AddLabelOnNode(node.Name, regionKey, val)
+			require.NoError(t, err)
+		}
+
+		// Install the cluster and do validations
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
+
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Add zone label to all nodes, now constraint changed so new pods should be created
+		for i, node := range nodeList.Items {
+			val := fmt.Sprintf("zone%v", i)
+			logrus.Infof("Label node %s with %s=%s", node.Name, zoneKey, val)
+			err := coreops.Instance().AddLabelOnNode(node.Name, zoneKey, val)
+			require.NoError(t, err)
+		}
+		logrus.Infof("Validate StorageCluster %s", cluster.Name)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster,
+			ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
+		require.NoError(t, err)
+
+		// Remove topology labels from all nodes and validate pod topology spread constraints
+		for _, node := range nodeList.Items {
+			ci_utils.RemoveLabelFromNode(t, node.Name, regionKey)
+			ci_utils.RemoveLabelFromNode(t, node.Name, zoneKey)
+		}
+		logrus.Infof("Validate StorageCluster %s", cluster.Name)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster,
+			ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
+		require.NoError(t, err)
+
+		// Uninstall storage cluster and recover labels
+		ci_utils.UninstallAndValidateStorageCluster(cluster, t)
+		logrus.Infof("Recovering cluster topology labels")
+		for _, node := range nodeList.Items {
+			if val, ok := node.Labels[regionKey]; ok {
```

Comment:
If the nodes don't have any topology labels, then we still want to remove them after the test, right? This won't handle that part.

Reply:
removing label is done as a test step before uninstalling the cluster, so at this step all topology labels are already removed.